types/types.proto

```diff
@@ -78,3 +79,177 @@ message SimulationResponse {
   GasInfo gas_info = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
   Result  result   = 2;
 }
+
+// Tx is the standard type used for broadcasting transactions
+message Tx {
```

Comment:
How do you feel about keeping the existing `Tx` interface and moving these types to `std`? Not blocking, but it's less of a breaking change.

Reply:
So I wrote a whole ADR on how to organize and name protobuf pakages (#6083). I would appreciate us doing a round of reviews on it sometime soon 😉 

But upon reflection, I actually think this shouldn't be in `types/` (nor should the simulation stuff). So I'll move it. Currently I think `types/tx` makes sense because it's not really the same meaning we had for `std` but I don't have a strong preference... Overall I'd prefer we agree on some guiding conventions and follow those, keeping in mind that proto names are intended to not change much

Reply:
Also something with our file layout is causing problems with protoc:

```
❯ make proto-gen
2020/05/13 19:03:43 protoc-gen-gogo: error:inconsistent package import paths: "github.com/cosmos/cosmos-sdk/types", "github.com/cosmos/cosmos-sdk/types/tx"
--gocosmos_out: protoc-gen-gocosmos: Plugin failed with status code 1.
```

I can't actually move it to `types/tx`. Maybe following https://buf.build/docs/lint-checkers#file_layout would help, not sure...

Reply:
I moved this stuff to `tx/types` which I think makes more sense. I'm seeing more tx related stuff coming down the pipeline (like new signing mode support), and maybe a top-level `tx` related package is the right home?

Reply:
Not sure I agree that having a root-level tx package makes sense. I'll need to think on this a bit...

Reply:
What do you think?

My main preference for now is that this stuff lives in its own package (so dependencies are isolated and it's easy to refactor). And if we go forward with #6083 we might reorganize more then.

Reply:
What is a solid argument for having this here opposed to `std/`?

Reply:
1. This is what we have as the doc string for `std`:
```
Package std defines all the common and standard inter-module Cosmos SDK types and definitions
modules and applications can depend on. These types and definitions serve as
convenient starting point and also to act as an example or template of how said
types and definitions can be overridden/redefined.

Typically these will be used in application-specific units of business logic. These
types and definitions are different from the types you may find in a "core" or "types"
package as those should be considered the defacto types to be used and are not
application-specific.
```

That may not be relevant anymore with the move to `Any` but that's a separate matter. Either way it doesn't convey that it's a place for `Tx`.

2. This `Tx` type is actually quite isolated from other types and modules, only depending on crypto primitives and coins. Therefore it deserves to be in a package that has clear small dependencies not a bag of `std` types

3. `Tx` likely deserves a protobuf package that relates to it's function like `tx`, not `std` which is non-descriptive

Reply:
Why not? We still have the `Tx` interface, do we not? And so, the proto-`Tx` concrete type is more-or-less application-specific, is it not?

Reply:
It can be, but I feel like we went to a fair amount of effort to make it generalizable. I don't think we should remove the interfaces, but I do think we should treat this as the de-facto `Tx` type. So much of the discussion we had was around simplifying client-side development which benefits from a single format. In that spirit, I think we want to avoid  as much as possible scenarios where apps feel like they need to override.

Reply:
Ok so in that case, why not have in `types/` like you originally had?

Reply:
Sure, I can revert. I wanted to have it in `types/tx` for better separation... because that won't work because of the protoc error (not the first time I've run into this)